server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -353,11 +363,63 @@ public short getDefaultReplication(Path path) {
     return getFileSystemByPath(path).getDefaultReplication(path);
   }
 
+  /**
+   * The Hadoop Configuration object does not currently allow for duplicate properties to be set in
+   * a single Configuration for different FileSystem URIs. Here we will look for properties in the
+   * Accumulo configuration of the form:
+   *
+   * <pre>
+   * instance.volume.config.&lt;volume-uri&gt;.&lt;hdfs-property&gt;
+   * </pre>
+   *
+   * We will use these properties to return a new Configuration object that can be used with the
+   * FileSystem URI.
+   *
+   * @param conf AccumuloConfiguration object
+   * @param hadoopConf Hadoop Configuration object
+   * @param filesystemURI Volume Filesystem URI
+   * @return Hadoop Configuration with custom overrides for this FileSystem
+   */
+  private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
+      final Configuration hadoopConf, final String filesystemURI) {
+    final Configuration volumeConfig = new Configuration(hadoopConf);
+    final Map<String,String> customProps =
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUMES_CONFIG);
+    customProps.forEach((key, value) -> {
+      if (key.startsWith(filesystemURI)) {
+        String property = key.substring(filesystemURI.length() + 1);
+        log.info("Overriding property {} to {} for volume {}", property, value, filesystemURI);
+        volumeConfig.set(property, value);
+      }
+    });
```

Comment:
This can be written slightly more succinctly without an intermediate map variable. This doesn't save much, but I do think there's some expressive value in separating the Predicate `if` clause from the `forEach` action:

```suggestion
    conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUMES_CONFIG).entrySet().stream()
        .filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
          String key = e.getKey().substring(filesystemURI.length() + 1);
          String value = e.getValue();
          log.info("Overriding property {} for volume {}", key, value, filesystemURI);
          volumeConfig.set(key, value);
        });
```

And again, you need to include the dot when checking for the startsWith... or you'll get false positives from similarly named paths.

Reply:
Addressed in f34eb45.